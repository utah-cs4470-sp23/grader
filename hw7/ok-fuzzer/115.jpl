let a = (({args, 68.83, (true != true)}{2} || (63 != (sum[a : to_int(30.87), b : 129, c : argnum] (- a)))) || ((to_float(argnum) <= 38.92) && (! (! true))))
let b = (78 + (sum[b : to_int(tan(atan(58.24))), c : (sum[b : (if (true == false) then (sum[b : 590, c : argnum, d : argnum] argnum) else 639), c : 572, d : (if (false && false) then (467 - argnum) else (argnum - argnum))] to_int(to_float(d))), d : (if false then {65.42, 17.41, argnum}{2} else (- 903))] d))
let c = (log((array[c : (if false then argnum else argnum), d : (argnum % b), e : b] to_float(argnum))[((b % b) / (if a then 619 else argnum)), 600, (sum[c : (sum[c : b] b), d : 631] (sum[e : argnum] b))]) + log(acos(sin(cos(99.5)))))
time print "d"


read image "d.png" to d[e, f]

let g[h] = [[58.46, {55.4, b, true}{0}], (array[g : 660] acos((- c))), {atan(c), (array[g : argnum] c)}{1}][844]
let {i[j, k], {l[m, n]}} = (array[i : to_int((- asin(c))), j : (- (- e))] {(if (j >= b) then (if false then d else d) else d), {(if false then d else d)}})[(argnum % (if ((h - b) > (if a then h else h)) then to_int(to_float(454)) else (if (h > e) then args[f] else b))), (if (array[i : (if a then b else b), j : h, k : e] (false || false))[(- to_int(c)), {533}{0}, b] then to_int(sqrt(pow(c, 12.42))) else (sum[i : (if (! a) then h else 911), j : h] (array[k : e, l : e, m : i] e)[(h + argnum), (j - e), i]))]
let o = (! (430 == (if (if a then true else (! false)) then {argnum, m, d}{1} else m)))
read image "p.png" to p

let {} = {}
write image l to "q.png"

let q[r] = (if a then [m] else args)
