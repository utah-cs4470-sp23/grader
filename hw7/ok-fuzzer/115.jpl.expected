(LetCmd
 (ArgLValue (VarArgument (BoolType) a))
 (BinopExpr
  (BoolType)
  (BinopExpr
   (BoolType)
   (TupleIndexExpr
    (BoolType)
    (TupleLiteralExpr
     (TupleType (ArrayType (IntType) 1) (FloatType) (BoolType))
     (VarExpr (ArrayType (IntType) 1) args)
     (FloatExpr (FloatType) 68)
     (BinopExpr (BoolType) (TrueExpr (BoolType)) != (TrueExpr (BoolType))))
    2)
   ||
   (BinopExpr
    (BoolType)
    (IntExpr (IntType) 63)
    !=
    (SumLoopExpr
     (IntType)
     a
     (CallExpr (IntType) to_int (FloatExpr (FloatType) 30))
     b
     (IntExpr (IntType) 129)
     c
     (VarExpr (IntType) argnum)
     (UnopExpr (IntType) - (VarExpr (IntType) a)))))
  ||
  (BinopExpr
   (BoolType)
   (BinopExpr
    (BoolType)
    (CallExpr (FloatType) to_float (VarExpr (IntType) argnum))
    <=
    (FloatExpr (FloatType) 38))
   &&
   (UnopExpr (BoolType) ! (UnopExpr (BoolType) ! (TrueExpr (BoolType)))))))
(LetCmd
 (ArgLValue (VarArgument (IntType) b))
 (BinopExpr
  (IntType)
  (IntExpr (IntType) 78)
  +
  (SumLoopExpr
   (IntType)
   b
   (CallExpr
    (IntType)
    to_int
    (CallExpr
     (FloatType)
     tan
     (CallExpr (FloatType) atan (FloatExpr (FloatType) 58))))
   c
   (SumLoopExpr
    (IntType)
    b
    (IfExpr
     (IntType)
     (BinopExpr (BoolType) (TrueExpr (BoolType)) == (FalseExpr (BoolType)))
     (SumLoopExpr
      (IntType)
      b
      (IntExpr (IntType) 590)
      c
      (VarExpr (IntType) argnum)
      d
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) argnum))
     (IntExpr (IntType) 639))
    c
    (IntExpr (IntType) 572)
    d
    (IfExpr
     (IntType)
     (BinopExpr (BoolType) (FalseExpr (BoolType)) && (FalseExpr (BoolType)))
     (BinopExpr (IntType) (IntExpr (IntType) 467) - (VarExpr (IntType) argnum))
     (BinopExpr
      (IntType)
      (VarExpr (IntType) argnum)
      -
      (VarExpr (IntType) argnum)))
    (CallExpr
     (IntType)
     to_int
     (CallExpr (FloatType) to_float (VarExpr (IntType) d))))
   d
   (IfExpr
    (IntType)
    (FalseExpr (BoolType))
    (TupleIndexExpr
     (IntType)
     (TupleLiteralExpr
      (TupleType (FloatType) (FloatType) (IntType))
      (FloatExpr (FloatType) 65)
      (FloatExpr (FloatType) 17)
      (VarExpr (IntType) argnum))
     2)
    (UnopExpr (IntType) - (IntExpr (IntType) 903)))
   (VarExpr (IntType) d))))
(LetCmd
 (ArgLValue (VarArgument (FloatType) c))
 (BinopExpr
  (FloatType)
  (CallExpr
   (FloatType)
   log
   (ArrayIndexExpr
    (FloatType)
    (ArrayLoopExpr
     (ArrayType (FloatType) 3)
     c
     (IfExpr
      (IntType)
      (FalseExpr (BoolType))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) argnum))
     d
     (BinopExpr (IntType) (VarExpr (IntType) argnum) % (VarExpr (IntType) b))
     e
     (VarExpr (IntType) b)
     (CallExpr (FloatType) to_float (VarExpr (IntType) argnum)))
    (BinopExpr
     (IntType)
     (BinopExpr (IntType) (VarExpr (IntType) b) % (VarExpr (IntType) b))
     /
     (IfExpr
      (IntType)
      (VarExpr (BoolType) a)
      (IntExpr (IntType) 619)
      (VarExpr (IntType) argnum)))
    (IntExpr (IntType) 600)
    (SumLoopExpr
     (IntType)
     c
     (SumLoopExpr (IntType) c (VarExpr (IntType) b) (VarExpr (IntType) b))
     d
     (IntExpr (IntType) 631)
     (SumLoopExpr (IntType) e (VarExpr (IntType) argnum) (VarExpr (IntType) b)))))
  +
  (CallExpr
   (FloatType)
   log
   (CallExpr
    (FloatType)
    acos
    (CallExpr
     (FloatType)
     sin
     (CallExpr (FloatType) cos (FloatExpr (FloatType) 99)))))))
(TimeCmd (PrintCmd "d"))
(ReadCmd "d.png" (ArrayArgument d e f))
(LetCmd
 (ArgLValue (ArrayArgument g h))
 (ArrayIndexExpr
  (ArrayType (FloatType) 1)
  (ArrayLiteralExpr
   (ArrayType (ArrayType (FloatType) 1) 1)
   (ArrayLiteralExpr
    (ArrayType (FloatType) 1)
    (FloatExpr (FloatType) 58)
    (TupleIndexExpr
     (FloatType)
     (TupleLiteralExpr
      (TupleType (FloatType) (IntType) (BoolType))
      (FloatExpr (FloatType) 55)
      (VarExpr (IntType) b)
      (TrueExpr (BoolType)))
     0))
   (ArrayLoopExpr
    (ArrayType (FloatType) 1)
    g
    (IntExpr (IntType) 660)
    (CallExpr
     (FloatType)
     acos
     (UnopExpr (FloatType) - (VarExpr (FloatType) c))))
   (TupleIndexExpr
    (ArrayType (FloatType) 1)
    (TupleLiteralExpr
     (TupleType (FloatType) (ArrayType (FloatType) 1))
     (CallExpr (FloatType) atan (VarExpr (FloatType) c))
     (ArrayLoopExpr
      (ArrayType (FloatType) 1)
      g
      (VarExpr (IntType) argnum)
      (VarExpr (FloatType) c)))
    1))
  (IntExpr (IntType) 844)))
(LetCmd
 (TupleLValue
  (ArgLValue (ArrayArgument i j k))
  (TupleLValue (ArgLValue (ArrayArgument l m n))))
 (ArrayIndexExpr
  (TupleType
   (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
   (TupleType
    (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)))
  (ArrayLoopExpr
   (ArrayType
    (TupleType
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)))
    2)
   i
   (CallExpr
    (IntType)
    to_int
    (UnopExpr
     (FloatType)
     -
     (CallExpr (FloatType) asin (VarExpr (FloatType) c))))
   j
   (UnopExpr (IntType) - (UnopExpr (IntType) - (VarExpr (IntType) e)))
   (TupleLiteralExpr
    (TupleType
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)))
    (IfExpr
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     (BinopExpr (BoolType) (VarExpr (IntType) j) >= (VarExpr (IntType) b))
     (IfExpr
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      (FalseExpr (BoolType))
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       d)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       d))
     (VarExpr
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      d))
    (TupleLiteralExpr
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2))
     (IfExpr
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      (FalseExpr (BoolType))
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       d)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       d)))))
  (BinopExpr
   (IntType)
   (VarExpr (IntType) argnum)
   %
   (IfExpr
    (IntType)
    (BinopExpr
     (BoolType)
     (BinopExpr (IntType) (VarExpr (IntType) h) - (VarExpr (IntType) b))
     >
     (IfExpr
      (IntType)
      (VarExpr (BoolType) a)
      (VarExpr (IntType) h)
      (VarExpr (IntType) h)))
    (CallExpr
     (IntType)
     to_int
     (CallExpr (FloatType) to_float (IntExpr (IntType) 454)))
    (IfExpr
     (IntType)
     (BinopExpr (BoolType) (VarExpr (IntType) h) > (VarExpr (IntType) e))
     (ArrayIndexExpr
      (IntType)
      (VarExpr (ArrayType (IntType) 1) args)
      (VarExpr (IntType) f))
     (VarExpr (IntType) b))))
  (IfExpr
   (IntType)
   (ArrayIndexExpr
    (BoolType)
    (ArrayLoopExpr
     (ArrayType (BoolType) 3)
     i
     (IfExpr
      (IntType)
      (VarExpr (BoolType) a)
      (VarExpr (IntType) b)
      (VarExpr (IntType) b))
     j
     (VarExpr (IntType) h)
     k
     (VarExpr (IntType) e)
     (BinopExpr (BoolType) (FalseExpr (BoolType)) || (FalseExpr (BoolType))))
    (UnopExpr (IntType) - (CallExpr (IntType) to_int (VarExpr (FloatType) c)))
    (TupleIndexExpr
     (IntType)
     (TupleLiteralExpr (TupleType (IntType)) (IntExpr (IntType) 533))
     0)
    (VarExpr (IntType) b))
   (CallExpr
    (IntType)
    to_int
    (CallExpr
     (FloatType)
     sqrt
     (CallExpr
      (FloatType)
      pow
      (VarExpr (FloatType) c)
      (FloatExpr (FloatType) 12))))
   (SumLoopExpr
    (IntType)
    i
    (IfExpr
     (IntType)
     (UnopExpr (BoolType) ! (VarExpr (BoolType) a))
     (VarExpr (IntType) h)
     (IntExpr (IntType) 911))
    j
    (VarExpr (IntType) h)
    (ArrayIndexExpr
     (IntType)
     (ArrayLoopExpr
      (ArrayType (IntType) 3)
      k
      (VarExpr (IntType) e)
      l
      (VarExpr (IntType) e)
      m
      (VarExpr (IntType) i)
      (VarExpr (IntType) e))
     (BinopExpr (IntType) (VarExpr (IntType) h) + (VarExpr (IntType) argnum))
     (BinopExpr (IntType) (VarExpr (IntType) j) - (VarExpr (IntType) e))
     (VarExpr (IntType) i))))))
(LetCmd
 (ArgLValue (VarArgument (BoolType) o))
 (UnopExpr
  (BoolType)
  !
  (BinopExpr
   (BoolType)
   (IntExpr (IntType) 430)
   ==
   (IfExpr
    (IntType)
    (IfExpr
     (BoolType)
     (VarExpr (BoolType) a)
     (TrueExpr (BoolType))
     (UnopExpr (BoolType) ! (FalseExpr (BoolType))))
    (TupleIndexExpr
     (IntType)
     (TupleLiteralExpr
      (TupleType
       (IntType)
       (IntType)
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) m)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       d))
     1)
    (VarExpr (IntType) m)))))
(ReadCmd
 "p.png"
 (VarArgument
  (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
  p))
(LetCmd (TupleLValue) (TupleLiteralExpr (TupleType)))
(WriteCmd
 (VarExpr
  (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
  l)
 "q.png")
(LetCmd
 (ArgLValue (ArrayArgument q r))
 (IfExpr
  (ArrayType (IntType) 1)
  (VarExpr (BoolType) a)
  (ArrayLiteralExpr (ArrayType (IntType) 1) (VarExpr (IntType) m))
  (VarExpr (ArrayType (IntType) 1) args)))
